util/argo/argo.go

```diff
@@ -115,6 +115,43 @@ func FilterByProjects(apps []argoappv1.Application, projects []string) []argoapp
 	return items
 }
 
+func FilterByPath(apps []argoappv1.Application, path string) []argoappv1.Application {
+	filteredApps := make([]argoappv1.Application, 0)
+	for _, app := range apps {
+		if app.Spec.Source != nil && (app.Spec.Source.Path == path || strings.HasPrefix(app.Spec.Source.Path, path+"/")) {
```

Comment:
```suggestion
		if app.Spec.Source != nil && (app.Spec.Source.Path == path || strings.HasPrefix(app.Spec.Source.Path, path+"/")) {
```
You'd also need to filter the apps that consist of multiple sources. Right now, it only checks for a single source application.